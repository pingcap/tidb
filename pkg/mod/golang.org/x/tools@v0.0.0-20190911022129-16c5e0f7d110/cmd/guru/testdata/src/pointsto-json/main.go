package main

// Tests of 'pointsto' queries, -format=json.
// See go.tools/guru/guru_test.go for explanation.
// See pointsto-json.golden for expected query results.

func main() { //
	var s struct{ x [3]int }
	p := &s.x[0] // @pointsto val-p "p"
	_ = p

	var i I = C(0)
	if i == nil {
		i = new(D)
	}
	print(i) // @pointsto val-i "\\bi\\b"
}

type I interface {
	f()
}

type C int
type D struct{}

func (c C) f()  {}
func (d *D) f() {}
