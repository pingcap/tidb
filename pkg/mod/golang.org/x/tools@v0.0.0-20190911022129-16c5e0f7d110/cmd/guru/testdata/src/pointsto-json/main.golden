-------- @pointsto val-p --------
[
	{
		"type": "*int",
		"labels": [
			{
				"pos": "testdata/src/pointsto-json/main.go:8:6",
				"desc": "s.x[*]"
			}
		]
	}
]
-------- @pointsto val-i --------
[
	{
		"type": "*D",
		"namepos": "testdata/src/pointsto-json/main.go:24:6",
		"labels": [
			{
				"pos": "testdata/src/pointsto-json/main.go:14:10",
				"desc": "new"
			}
		]
	},
	{
		"type": "C",
		"namepos": "testdata/src/pointsto-json/main.go:23:6"
	}
]
